### PR TITLE
Fix composer.lock warning after semantic release

### PR DESCRIPTION
**Pull request by Jonnx**
# Changes

This PR resolves the warning generated by composer when a semantic version is committed, the code is checked out and the `composer install` command is ran.

```
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>
```
This PR resolves this issue by updating the composer.lock file after the semantic version has been written to the composer.json file using the command `composer update --lock`. So the execution of the semantic release script is as follows:

1. semantic-release-composer updates the composer.json file
2. semantic-release-composer checks that a composer.lock file exists (new)
3. semantic-release-composer runs `composer update --lock` which modifies the composer.lock file (new) 

## Notes on `composer update --lock`

It appears as this command does not update any dependencies as part of its execution, which is desired in this case as we do not want to modify any locked versions of installed packages and just want to keep the composer.lock in sync with the composer.json files.

Source: https://getcomposer.org/doc/03-cli.md#main

```
--lock: Only updates the lock file hash to suppress warning about the lock file being out of date.
```

## Related tickets

- https://github.com/ambimax/semantic-release-composer/issues/14

## PR TODO

- [x] I have added required information to README.md
- [x] I have performed a self-review of my own code
- [x] I have updated and referenced related tickets
- [x] My commits contain [Semantic Release syntax](https://guide.ambimax.xyz/best-practices/semantic-release/#git-commits)

### CODE REVIEW

**Comment by Jonnx** [comment]
Thoughts on letting CI run and merging this in? Otherwise I can setup my own fork and CI, but I would prefer to keep this with the original project.

**Comment by Jonnx** [comment]
if anyone is running into this issue, and wants to use my fork while this pr / issue is awaiting response, you can update you package.json file as follows and you will receive the patched version

```
{
  ...
  "devDependencies": {
      "@ambimax/semantic-release-composer": "jonnx/semantic-release-composer",
      ...
  }
  ...
}
```

Running `yarn` then ensures the latest version is installed on your local machine if your node_modules are already installed locally

**Review by FabianKoehnen — APPROVED** [review]
Thanks for your contribution. Sorry for the long wait for approval

**Comment by Jonnx** [comment]
> Thanks for your contribution. Sorry for the long wait for approval

no problem! glad it got merged even though y'all are busy with your day jobs.

**Pull request merged**
